beluga_ros/tools/conversion_utils.py

```diff
@@ -0,0 +1,273 @@
+                yy,
+                ndt.get_scipy_representation().pdf(pos),
+                [0.1, 0.5],
+                alpha=0.2,
+                linewidths=3,
+                extend="max",
+            )
+        if show:
+            plt.show()
+        return plt.gcf()
+
+    def to_hdf5(self, output_file_path: Path):
+        """
+        Serialize the NDT map into an HDF5 format.
+
+        See https://docs.hdfgroup.org/hdf5/develop/_intro_h_d_f5.html for details on the format.
+        It'll create 4 datasets:
+            - "resolution": () resolution for the discrete grid (cells are resolution x
+              resolution m^2 squares).
+            - "discrete_cells": (NUM_CELLS, 2) that contains the discrete cell coordinates.
+            - "means": (NUM_CELLS, 2) that contains the 2d mean of the normal distribution
+              of the cell.
+            - "covariances": (NUM_CELLS, 2, 2) Contains the covariance for each cell.
+        """
+        assert output_file_path.suffix in (".hdf5", ".h5")
+        output_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+        num_cells = len(self._grid.keys())
+
+        with h5py.File(output_file_path.absolute(), "w") as fp:
+            discrete_cells_dataset = fp.create_dataset(
+                "discrete_cells", (num_cells, 2), chunks=True
+            )
+            for idx, cell in enumerate(self._grid.keys()):
+                discrete_cells_dataset[idx] = np.asarray([cell.x, cell.y])
+
+            means_dataset = fp.create_dataset("means", (num_cells, 2), chunks=True)
+            covariances_dataset = fp.create_dataset("covariances", (num_cells, 2, 2))
+
+            for idx, distribution in enumerate(self._grid.values()):
+                means_dataset[idx] = distribution.mean
+                covariances_dataset[idx] = distribution.covariance
+            fp.create_dataset("resolution", data=np.asarray(self._resolution))
+
+    @classmethod
+    def from_hdf5(cls, hdf5_file: Path):
+        """
+        Create an NDTMap instance from a path to a hdf5 file.
+
+        See 'to_hdf5' docstring for details on the hdf5 format.
+        """
+        with h5py.File(hdf5_file.absolute(), "r") as fp:
+            resolution: float = fp["resolution"][()]
+            means: np.ndarray = fp["means"]
+            discrete_cells: np.ndarray = fp["discrete_cells"]
+            covs: np.ndarray = fp["covariances"]
+            total_cells = covs.shape[0]
+
+            assert covs.shape[1] == 2
+            assert covs.shape[2] == 2
+            assert means.shape[1] == 2
+            assert discrete_cells.shape[0] == total_cells
+            assert discrete_cells.shape[1] == 2
+            assert means.shape[0] == total_cells
+
+            ret = NDTMap(resolution=resolution)
+            for cell_num in range(total_cells):
+                cell = DiscreteCell(*discrete_cells[cell_num].tolist())
+                mean = means[cell_num]
+                cov = covs[cell_num]
```

Comment:
@serraramiro1 nit, fyi: `zip`'ing may be handy.

Reply:
100% right, done!